dsl/src/commonMain/kotlin/it/unibo/collektive/AggregateResult.kt

```diff
@@ -0,0 +1,18 @@
+package it.unibo.collektive
+
+import it.unibo.collektive.messages.SentMessage
+import it.unibo.collektive.stack.Path
+
+/**
+ * Result of the aggregate computation.
+ * @param localId: id of the device.
+ * @param result: result of the computation.
+ * @param toSend: map with id of the neighbour and relative message to send.
+ * @param newState: new state of the device.
+ */
+data class AggregateResult<X>(
+    val localId: ID,
+    val result: X,
+    val toSend: Set<SentMessage>,
+    val newState: Map<Path, *>, // todo create state data class
```

Comment:
Address this TODO

Reply:
I have now introduced the concept of State data class.